text/0000-deprecation.md

```diff
@@ -0,0 +1,158 @@
+- Feature Name: A plan for deprecating APIs within Rust
+- Start Date: 2015-06-03
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+
+There has been an ongoing [discussion on 
+internals](https://internals.rust-lang.org/t/thoughts-on-aggressive-deprecation-in-libstd/2176/55) 
+about how we are going to evolve the standard library. This RFC tries 
+to condense the consensus.
+
+As a starting point, the current deprecation feature allows a developer
+to annotate their API items with `#[deprecated(since="1.1.0")]` and
+have suitable warnings shown if the feature is used.
+
+# Motivation
+
+We want to:
+
+1. evolve the `std` API, including making items unavailable with new 
```

Comment:
I definitely agree that we want to evolve the APIs that libstd provides, but I'm not quite in agreement that we want to *remove* deprecated APIs. I think that the overhead of keeping around unstable APIs is unlikely to become too burdensome, but this is of course tough to predict. I also think that it's possible to leave these APIs around without confusing new users through various mitigation tactics like compiler error messages, documentation, etc.